### ISSUE TITLE
Fx 4.0b2pre: "Get Add-ons" page blocked in about:addons

### ISSUE THREAD

**Issue by jsamuel** [RequestPolicy/requestpolicy#126]
```
imported trac ticket
created: 2010-07-13 10:07:07
reporter: justin
```

When one opens about:addons and clicks on the "Get Add-ons" section, the request is blocked and nothing shows up:

{{{
[RequestPolicy] [INFO] [INTERNAL] Unable to getIdentifier from uri about:addons using identifier level 1.
[RequestPolicy] [WARNING] [CONTENT] *\* BLOCKED *\* reason: hosts don't match. type: 7, destination: https://services.addons.mozilla.org/en-US/firefox/discovery/4.0b2pre/Linux, origin: about:addons, context: [object XPCNativeWrapper [object XULElement]], mime: , null
}}}

**Comment by jsamuel** [comment]
```
imported trac comment
created: 2010-07-13 10:09:46
author: justin
```

Thanks to Aleksej for noticing this, and for generally being awesome in testing and finding bugs with RequestPolicy on Firefox 4.

**Comment by jsamuel** [comment]
```
imported trac comment
created: 2010-08-24 13:44:22
author: justin
```

I don't consider this as bad of an issue as I first thought. The user is shown the blocked request to mozilla.com and can allow it if they desire to do so. I'm not a big fan of a browser being embedded in about:addons like an iframe when you click on "Get Add-ons". It seems like a bad move that could result in users wandering around to non-mozilla pages and not being able to tell what site they are at because the location bar says "about:addons" the whole time. With any luck this will be replaced with something better before Fx 4 and it will become a non-issue.

**Comment by jsamuel** [comment]
```
imported trac comment
created: 2011-01-07 09:43:43
author: justin
```

Fixed in r383.

Fortunately, the iframe to the mozilla site has been changed so that all links within that iframe are opening in a new tab. This reduces the concern over users browsing the web in that iframe without their location bar changing.

In r383, I've added all origin/destination pairs for the iframe content to the permanent whitelist ("compatibility rules") and users should see the content of iframe without having to allow any requests through RP. It's possible that at some point mozilla will change the page shown in the iframe so that it relies on content from different domains than when I looked at it now. It wouldn't be a bad idea to occasionally check to see if any requests are being blocked.
